src/network/taraxa_capability.cpp

```diff
@@ -66,19 +66,38 @@ std::shared_ptr<TaraxaPeer> TaraxaCapability::getPeer(NodeID const &node_id) {
   return nullptr;
 }
 
+std::shared_ptr<TaraxaPeer> TaraxaCapability::getPendingPeer(NodeID const &node_id) {
+  boost::shared_lock<boost::shared_mutex> lock(peers_mutex_);
+  auto itPeer = pending_peers_.find(node_id);
+  if (itPeer != pending_peers_.end()) {
+    return itPeer->second;
+  }
+  return nullptr;
+}
+
 unsigned int TaraxaCapability::getPeersCount() {
   boost::shared_lock<boost::shared_mutex> lock(peers_mutex_);
   return peers_.size();
 }
 
 void TaraxaCapability::erasePeer(NodeID const &node_id) {
   boost::unique_lock<boost::shared_mutex> lock(peers_mutex_);
+  pending_peers_.erase(node_id);
   peers_.erase(node_id);
 }
 
 std::shared_ptr<TaraxaPeer> TaraxaCapability::insertPeer(NodeID const &node_id) {
   boost::unique_lock<boost::shared_mutex> lock(peers_mutex_);
-  auto ret = peers_.emplace(std::make_pair(node_id, std::make_shared<TaraxaPeer>(node_id)));
+  auto ret = pending_peers_.emplace(std::make_pair(node_id, std::make_shared<TaraxaPeer>(node_id)));
+  assert(ret.second);
+  return ret.first->second;
+}
+
+std::shared_ptr<TaraxaPeer> TaraxaCapability::insertPendingPeer(NodeID const &node_id,
```

Comment:
maybe it is just me, but naming is kind of confusing
 `insertPeer` -> inserting to `pending_peers_` and `insertPendingPeer`-> inserting to `peers_` ?

Reply:
Good point @MatusKysel !   That is indeed confusing ... although I get how it happened. :-)

Reply:
I added new names for both methods.